provider/kubernetes/kubernetes.go

```diff
@@ -53,7 +73,7 @@ type Provider struct {
 	lastConfiguration      safe.Safe
 }
 
-func (p *Provider) newK8sClient() (Client, error) {
+func (p Provider) newK8sClient() (Client, error) {
```

Comment:
Do we really want a copy of `Provider` when calling `newK8sClient` now?

Reply:
We don't reload the static configuration (eg we don't recreate the client).

`NewInClusterClient(endpoint string)` and `NewExternalClusterClient(endpoint, token, caFilePath string)` use string, a pointer as receiver don't change anything.

The use of a pointer as a receiver must be reserved only when a method changes the state of the receiver.

Maybe I have miss something?

Reply:
Late to the game, sorry.

The problem with receivers in Go is that they conflate two aspects: Mutability and performance / copy overhead. The [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments#receiver-type) have a paragraph on the subject, but eventually suggest to leans towards pointer receivers, especially when the involved receiver is a non-trivial struct. I certainly think that `Provider` is rather complex (especially since it embeds another `BaseProvider`), so a pointer receiver would have been in order for me.